StreamChatClient_v3/Controllers/ChannelListController.swift

```diff
@@ -134,6 +134,52 @@ public class ChannelListControllerGeneric<ExtraData: ExtraDataTypes>: Controller
     }
 }
 
+// MARK: - Actions
+
+public extension ChannelListControllerGeneric {
+    /// Delete the channel.
+    /// - Parameters:
+    ///   - channel: The channel you want to delete.
+    ///   - completion: An empty completion block.
+    func delete(channel: ChannelModel<ExtraData>, completion: ((Error?) -> Void)? = nil) {
+        client.apiClient.request(endpoint: .delete(channel: channel)) { result in
+            switch result {
+            case .success: completion?(nil)
+            case .failure(let error): completion?(error)
+            }
+        }
```

Comment:
This should live inside `ChannelUpdater`. It might seem like an additional indirection but it's about code reusability. `ChannelUpdater` won't be used only by this controller.